tasks/main.yml

```diff
@@ -1,27 +1,45 @@
 # SPDX-License-Identifier: MIT
 ---
-# Put the tasks for your role here.
-
 - name: Set platform/version specific variables
   include_tasks: tasks/set_vars.yml
 
-# Examples of some tasks:
 - name: Ensure required packages are installed
   package:
-    name: "{{ __template_packages }}"
+    name: "{{ __journald_packages }}"
     state: present
 
-- name: Ensure required services are enabled and started
+- name: Ensure required services are enabled
   service:
     name: "{{ item }}"
     state: started
     enabled: true
-  loop: "{{ __template_services }}"
+  loop: "{{ __journald_services }}"
+
+- name: Create journald configuration drop-in directory
+  file:
+    path: "{{ __journald_dropin_dir }}"
+    state: directory
+    owner: root
+    group: root
+    mode: '0755'
+
+- name: Configure persistent journal
+  include_tasks: tasks/persistent.yml
+  when: journald_persistent | bool
 
-- name: Generate /etc/{{ __template_foo_config }}
+- name: Configure volatile journal
+  include_tasks: tasks/volatile.yml
+  when: not (journald_persistent | bool)
+
+- name: Deploy journald configuration drop-in
   template:
-    src: "{{ __template_foo_config }}.j2"
-    dest: /etc/{{ __template_foo_config }}
-    backup: true
-    mode: 0400
-  notify: Template handler to restart services
+    src: journald.conf.j2
+    dest: "{{ __journald_dropin_dir }}/{{ __journald_dropin_conf }}"
+    owner: root
+    group: root
+    mode: '0644'
+  notify:
+    - Restart services
+
+- name: Flush handlers
+  meta: flush_handlers
```

Comment:
This will run *every handler in the playbook*.  If the user has many roles in the playbook, then all of them will get triggered here - probably not what the user is expecting.

Ansible will automatically run the handlers when "it is convenient for Ansible".  And maybe this is good enough for the journald role.  However, if you need finer control - if you want the service restarted immediately after creating or changing the configuration - then you will need to do that "manually".  Basically - figure out if tasks/persistent.yml or tasks/volatile.yml or the `Deploy journald configuration drop-in` task changed something - register some variable(s) - then have the last task like this:
```yaml
- name: Restart journald services
  service:
    name: "{{ item }}"
    state: restarted
  loop: "{{ __journald_services }}"
  when: __journald_persistent is changed or __journald_volatile is changed or __journald_config is changed
```

Reply:
Right, I don't necessarily need to flush the handlers here, I will just drop it.